astroid/brain/brain_namedtuple_enum.py

```diff
@@ -133,7 +133,11 @@ def infer_func_form(
     # we know it is a namedtuple anyway.
     name = name or "Uninferable"
     # we want to return a Class node instance with proper attributes set
-    class_node = nodes.ClassDef(name, parent=node.parent)
+    class_node = nodes.ClassDef(name)
+    # A typical ClassDef automatically adds its name to the parent scope,
+    # but doing so causes problems, so defer setting parent until after init
+    # see: https://github.com/PyCQA/pylint/issues/5982
+    class_node.parent = node.parent
```

Comment:
I borrowed the wording from: 
https://github.com/PyCQA/astroid/blob/16322e84801fc143906db33868abf5813875ae30/astroid/objects.py#L265-L267